### PR TITLE
chore: bump DS rc-7

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

1. Design system and icons bumped to rc-7 release
2. Grid items, buttons and Icon buttons styles adjusted.

Related RC release changes - https://github.com/strapi/design-system/pull/1746

### CODE REVIEW

**Comment by lucasboilly** [comment]
Hello, I noticed a few things:

1. In the tables, we're supposed to use the "ghost" version of the icon button (the one with no outline)

CleanShot 2024-07-23 at 12 22 37@2x

CleanShot 2024-07-23 at 12 23 05@2x

2. The icon to sort the table became a default icon button but it's too big. Would it be the right moment to introduce a new smaller size for it?

CleanShot 2024-07-23 at 12 23 34@2x

3. The icons in the popovers became Neutral 800 but they're supposed to be Neutral 500.

CleanShot 2024-07-23 at 12 25 43@2x

CleanShot 2024-07-23 at 12 36 09@2x

4. An outline appeared here in the configure the view popover.

CleanShot 2024-07-23 at 12 26 21@2x

Apart from that, all other icon buttons now have the right size 👍

**Comment by madhurisandbhor** [comment]
@lucasboilly first and second issues are already fixed could you please try clearing cache or in different browser, thanks

**Comment by lucasboilly** [comment]
@madhurisandbhor I cleared cache and tested on multiple other browsers (Firefox, Safari and Chrome) but I'm still facing the same issues.

In the meantime I also noticed an issue with the expand/collapse button of the old RTE:


https://github.com/user-attachments/assets/71535796-26b9-40cc-abc6-5fcf641b0145

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx [hunk 1]
```diff
@@ -298,7 +300,14 @@ const DocumentActionsMenu = ({
                   gap={2}
                   tag="span"
                 >
-                  <span>{action.icon}</span>
+                  <Flex
+                    tag="span"
+                    color={
+                      !action.disabled ? convertActionVariantToIconColor(action.variant) : 'inherit'
```
Comment (remidej) [inline packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:306]: That logic to get the color is the same as in the one in the parent Flex above. So if you set inherit as the color here, regardless of the disabled state, don't you get the same result?

**Review by remidej — COMMENTED** [review]
I looked at the changes and they generally look to me! There's no way I can review every single change on such a big diff though, so I think we'll have to rely heavily on manual QA before we merge this one

File: packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx (hunk 1 above)
Comment (madhurisandbhor) [reply packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx:306]: no its undefined but secondary where it was breaking, thats why we have `convertActionVariantToIconColor` as a separate function.

**Review by remidej — COMMENTED** [review]
I checked [Lucas's Loom video](https://www.loom.com/share/7236f88cf0234e3e8b1376c9bfccbae1) and I'm seeing the same small issues as him. Except the configure the view button, it looks fine on my end.

I notice something else though, in the main nav the click boxes aren't square anymore, they're slightly too tall

**Comment by madhurisandbhor** [comment]
@remidej please post images of new things you found.

**Comment by madhurisandbhor** [comment]
For the links hover style, I have raised a [PR](https://github.com/strapi/design-system/pull/1765) in DS but it will be available in next DS RC-8 release.

**Comment by lucasboilly** [comment]
Hey @madhurisandbhor and @remidej 

Here's what I found:

Corrected:
- Icons in IconButtons are centered
- Items of the MainNav are squares

Not corrected:
- Blue hover states (will be fixed in RX-8)
- Focused state of some IconButtons with a black outline
- IconButtons are still too big

New issues:
- Kind of bug in the animation of the overlay behind the modals when opened
https://github.com/user-attachments/assets/1ed2edc4-fb99-4499-942c-83c8618242e5
- Missing border radius on some popovers
CleanShot 2024-07-30 at 10 49 14@2x

**Comment by madhurisandbhor** [comment]
@lucasboilly Thank you for taking time to test again!! 🙌 

> Not corrected:
> 
> Blue hover states (fixed in RC-8)
> Focused state of some IconButtons with a black outline - https://github.com/strapi/design-system/issues/1767
> IconButtons are still too big (fixed in RC-8)

Focused state of some IconButtons with a black outline -  will be fixed in RC-8 https://github.com/strapi/design-system/issues/1767


> New issues:
> 
> The help button on the bottom right corner is not round anymore 

fixed in this PR
 
> Kind of bug in the animation of the overlay behind the modals when opened 
 
 not related to this change, will report it
 
> Missing border radius on some popovers

 will be fixed in RC-8 https://github.com/strapi/design-system/issues/1768

**Comment by lucasboilly** [comment]
@madhurisandbhor 

All good for the help button, it's round again 👍

I'm waiting for RC-8 to check the rest.

I noticed 3 other things (maybe it will be solved in RC-8 or later): 
- The click area of the inputs is a bit too small, you can see it when a field is highlighted by 1Password
CleanShot 2024-07-30 at 16 00 48@2x
- The icons in inputs are not aligned vertically
CleanShot 2024-07-30 at 16 13 40@2x
- A stretched button in the email configuration settings
CleanShot 2024-07-30 at 16 09 28@2x

**Comment by lucasboilly** [comment]
All non DS-related issues have been fixed 👍

**Review by remidej — APPROVED** [review]

**Pull request merged**